File: FUNDAMENTALS/35hat.py
import random

class Hat:
    # def __init__(self):
    #     self.houses = ["Shah Paran", "Mujtaba", "Bijoy 24"]
    #     self.names = ["Shihab1", "Radia", "Ali", "Shohan", "Rashed", "Rafsan"]
        
    houses = ["Shah Paran", "Mujtaba", "Bijoy 24"]
    # def sort(self, name):
    #     print(random.choice(self.names), 'is in', random.choice(self.houses))
    @classmethod
    def sort(cls, name):
        print(name, 'is in', random.choice(cls.houses))
    
# hat = Hat()

# i = input("How many times")
# for _ in range(int(i)):
#     hat.sort("Shihab")
Hat.sort("Shihab")

File: FUNDAMENTALS/35hat.cpp
#include <iostream>
#include <vector>
#include <string>
#include <cstdlib>
#include <ctime>

using namespace std;

class Hat {
public:
    vector<string> houses;

    Hat() {
        houses = {"Shah Paran", "Mujtaba", "Bijoy 24"};
    }

    void sort(const string& name) {
        int index = rand() % houses.size();
        cout << name << " is in " << houses[index] << endl;
    }
};

int main() {
    srand(time(0)); // seed random generator

    Hat hat;
    int i;

    cout << "How many times: ";
    cin >> i;

    for (int j = 0; j < i; j++) {
        hat.sort("Shihab");
    }

    return 0;
}
